File: server/src/main/java/dev/slimevr/vr/Device.java
package dev.slimevr.vr;

import dev.slimevr.vr.trackers.Tracker;

import java.net.InetAddress;
import java.util.HashMap;
import java.util.concurrent.atomic.AtomicInteger;


public class Device {

	public static final AtomicInteger nextLocalDeviceId = new AtomicInteger();

	private final int id;
	private String customName;
	private String firmwareVersion;
	private String manufacturer;

	private final HashMap<Integer, Tracker> trackers = new HashMap<>();

	public Device() {
		this.id = nextLocalDeviceId.incrementAndGet();
	}

	public int getId() {
		return id;
	}

	public void setCustomName(String customName) {
		this.customName = customName;
	}

	public String getManufacturer() {
		return this.manufacturer;
	}

	public void setManufacturer(String manufacturer) {
		this.manufacturer = manufacturer;
	}

	public String getDisplayName() {
		return null;
	}

	public String getFirmwareVersion() {
		return this.firmwareVersion;
	}

	public void setFirmwareVersion(String firmwareVersion) {
		this.firmwareVersion = firmwareVersion;
	}

	public String getCustomName() {
		return this.customName;
	}

	public InetAddress getIpAddress() {
		return null;
	}

	public HashMap<Integer, Tracker> getTrackers() {
		return trackers;
	}
}

File: server/src/main/java/dev/slimevr/tracking/trackers/Device.kt
package dev.slimevr.tracking.trackers

import dev.slimevr.tracking.trackers.udp.BoardType
import dev.slimevr.tracking.trackers.udp.MCUType
import java.util.concurrent.atomic.AtomicInteger

abstract class Device {
	open val id: Int = nextLocalDeviceId.incrementAndGet()
	open var name: String? = null
	open var firmwareVersion: String? = null
	open var manufacturer: String? = null
	open val trackers = HashMap<Int, Tracker>()

	/**
	 * Implement toString() to return a string that uniquely identifies the board type
	 * SHOULDN'T RETURN NULL WHEN toString() IS CALLED
	 */
	open val boardType: Any = BoardType.UNKNOWN
	open val mcuType: MCUType = MCUType.UNKNOWN

	abstract val hardwareIdentifier: String

	companion object {
		@JvmField
		val nextLocalDeviceId = AtomicInteger()
	}
}
